### ISSUE TITLE
Database.GetContainer(string) should return null when container is not found

### ISSUE THREAD

**Issue by leopasta** [Azure/azure-cosmos-dotnet-v3#590]
**Is your feature request related to a problem? Please describe.**
There should be a simple way to check if a container exists in a database.
Currently 
```
Container myContainer = db.GetContainer("MyFakeCollection")
```
returns a reference that looks valid, until you try an operation on it, when an exception is thrown. This is documented, but not a great implementation IMO.

**Describe the solution you'd like**
Either db.GetContainer("MyFakeCollection") would return null when it doesn't find a container or there should be a method in the container class to check if it exists in the db.

**Describe alternatives you've considered**
Try an operation and catch the exception - I don't consider this a good pattern when you expect the operation to fail often.

My least bad alternative is to create a method scan through all collections, but that should not be necessary:
```
        private Container GetContainer(Database db, string container)
        {
            FeedIterator<ContainerProperties> resultSetIterator = db.GetContainerQueryIterator<ContainerProperties>();
            while (resultSetIterator.HasMoreResults)
            {
                foreach (ContainerProperties candidate in resultSetIterator.ReadNextAsync().Result)
                {
                    if (candidate.Id == container)
                        return db.GetContainer(container);
                }
            }
            return null; 
        }
```

**Comment by j82w** [comment]
I think there is some confusion around the new v3 model. Database and Container are client side references. They are essentially creating and storing the URI to the resource. This allow you to pass around a single object instead of needing to pass around the client and the container URI info. As the [documentation](https://github.com/Azure/azure-cosmos-dotnet-v3/blob/c617ac285fec49a5980282d42c02176db45d2b18/Microsoft.Azure.Cosmos/src/Resource/Database/Database.cs#L235) says it does not validate the resource exists in Cosmos. You can then do operations off of that reference object like myContainer.ReadContainerAsync()

Why don't we validate? It requires doing a network call on a master resource which has a limited number of RUs. If you have the application installed on 100 vms and they are all using the same Container then it doesn't make since that every time the application starts you should need to validate the container exists.

If you want to validate the container exists there is a few different options. I would avoid doing query as it is rather expensive and slow compared to a single point operation.
1. [CreateContainerIfNotExistsAsync](https://github.com/Azure/azure-cosmos-dotnet-v3/blob/c617ac285fec49a5980282d42c02176db45d2b18/Microsoft.Azure.Cosmos/src/Resource/Database/Database.cs#L403)

`Container container = await db.CreateContainerIfNotExistsAsync(candidate.Id, candidate.PartitionKeyPath);`

2. Do a [ReadContainerAsync ](https://github.com/Azure/azure-cosmos-dotnet-v3/blob/c617ac285fec49a5980282d42c02176db45d2b18/Microsoft.Azure.Cosmos/src/Resource/Container/Container.cs#L73)after creating the Container reference object.
```
Container container = db.GetContainer(candidate.Id);
try{
container = await container.ReadyContainerAsync();
catch (CosmosException ex) when (ex.StatusCode == HttpStatusCode.NotFound){
   throw new ArgumentException("Container does exist")
}
```

**Comment by leopasta** [comment]
Yeah, I will do the operation and catch the exception, but still find the current implementation not user friendly and a downgrade from the collection of containers that was available in preview.
```
dbClient.Databases[dbName].Containers[containerName]
```
IMO it is almost a convention that a "getContainer" verb means you are fetching the resource/object. If it was called getContainerReference(...) it would be pretty unambiguous.

**Comment by j82w** [comment]
That's valid feedback. We might be able to do an alias to avoid a breaking change. 

We use Get verb for client operations, and Read for network calls. Also all network call end with Async. We moved away from index overloads because in our user testing no one ever discovered it which always caused confusion.

There is a shortcut on the client to get the container. 
`dbClient.GetContainer(dbName, containerName);`

**Comment by bartelink** [comment]
If its simply too late to change, I appreciate that, but this definitely feels wrong... Can the `Get` prefixing not just go? as mentioned on other issues in the builder arena, it's often doing a Create internally etc.

I know you guys are likely working hard to use it consistently, but the bottom line is that nobody used `Get` as the FDG (for good reason) recommends against it IIRC.
